src/NuGet.Services.AzureSearch/WebExceptionRetryDelegatingHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class WebExceptionRetryDelegatingHandler : DelegatingHandler
+    {
+        private readonly ILogger<WebExceptionRetryDelegatingHandler> _logger;
+
+        public WebExceptionRetryDelegatingHandler(ILogger<WebExceptionRetryDelegatingHandler> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public static HashSet<WebExceptionStatus> TransientWebExceptionStatuses => new HashSet<WebExceptionStatus>(new[]
```

Comment:
Does this need to be public? It looks like your tests have a copy of this. Also can this be a static field instead of a property (to avoid allocating on each access)?

Reply:
Yup, good catch. This should be a `static` and either a `readonly` property or field. Field is easier for `private` so I will go with that.